aclogview/CM_Communication.cs

```diff
@@ -176,6 +231,96 @@ public override void contributeToTreeView(TreeView treeView)
         }
     }*/
 
+    public class Emote : Message
+    {
+        public PStringChar emoteMessage;
+
+        public static Emote read(BinaryReader binaryReader)
+        {
+            var newObj = new Emote();
+            newObj.emoteMessage = PStringChar.read(binaryReader);
+            return newObj;
+        }
+
+        public override void contributeToTreeView(TreeView treeView)
+        {
+            TreeNode rootNode = new TreeNode(this.GetType().Name);
+            rootNode.Expand();
+            rootNode.Nodes.Add("emoteMessage = " + emoteMessage);
+            treeView.Nodes.Add(rootNode);
+        }
+    }
+
+    public class HearEmote : Message
+    {
+        public PStringChar EmoteMessage;
+        public PStringChar SenderName;
+        public uint SenderID;

```

Comment:
Please reorder the declarations and order of nodes in the tree view to match the read order